pkg/rancher-desktop/backend/wsl.ts

```diff
@@ -782,6 +783,28 @@ export default class WSLBackend extends events.EventEmitter implements VMBackend
     }
   }
 
+  /**
+   * Return the Linux path to the moproxy executable.
+   */
+  protected getMoproxyPath(): Promise<string> {
+    return this.wslify(path.join(paths.resources, 'linux', 'internal', 'moproxy'));
+  }
+
+  protected async writeProxySettings(proxy: BackendSettings['experimental']['virtualMachine']['proxy']): Promise<void> {
+    if (proxy.address && proxy.port) {
+      // Write to /etc/moproxy/proxy.ini
+      const protocol = proxy.address.startsWith('socks5://') ? 'socks5' : 'http';
+      const address = proxy.address.replace(/(https|http|socks5):\/\//g, '');
+      const contents = `[rancher-desktop-proxy]\naddress=${ address }:${ proxy.port }\nprotocol=${ protocol }\n`;
+      const username = proxy.username ? `http username=${ proxy.username }\n` : '';
```

Comment:
Will this work if the protocol is set to `socks5` on line `796`? or do we need an additional condition here too, to change to `socks username` and `socks password` if the protocol is set to `socks5`.

Reply:
Oops Indeed, I missed that.